ultralytics/models/yolo/model.py

```diff
@@ -43,3 +43,41 @@ def task_map(self):
                 "predictor": yolo.obb.OBBPredictor,
             },
         }
+
+
+class YOLOWorld(Model):
```

Comment:
@Laughing-q I noticed the new YOLOWorld class is pretty lightweight, could we implement these features directly in the YOLO class? This would probably be simpler for the users and allow us to simplify the code a bit?

Reply:
yeah I thought about the same thing, in fact I've implemented this idea but reverted it eventually. 
I figured we'll implement trainer for yolo-world(which definitely would be different from current yolo trainer) and in that case we'll need the YOLO class to handle both trainers of regular detector and open-vocabulary detector which seems to hard to implement and not clean. So I eventually create a YOLOWorld class.

BTW actually current YOLO class is good to directly create a yolo-world weight, the following code works properly:
```python
YOLO("yolov8s-world.pt")
```
but it'd fail to create a yolo-world model from scratch.

Reply:
@Laughing-q yeah that's interesting. How would the training process work for YOLO-World? Would it be YOLO training without any modifications to Clip?

I think in general we have a problem with Class loading. We should think about how to provide a common class capable of directing/loading any type of model. This would assist with CLI commands for other models too, which now run on over simplified logic, i.e. to load RTDETR from CLI the filename needs to include "rtdetr".

Ideally we'd want to migrate towards something like:

```python
from ultralytics import Model

model = Model('rtdetr.pt')
model = Model('sam.pt')
```

or we could migrate YOLO towards this behavior instead:
```python
from ultralytics import YOLO

model = YOLO('rtdetr.pt')
model = YOLO('sam.pt')
```

For this one let me think about this a little bit.

Reply:
I think in my example above Model() could work like this, except with more complicated decision logic that actually looks as the model architecture instead of just the filename:

```python
class Model:
    """A factory class for dynamically handling different model types in the Ultralytics package."""
    
    def __init__(self, model_path):
        """Initializes and loads the model, delegating to the appropriate class based on the model type."""
        self.model = self._load_model(model_path)
    
    def _load_model(self, model_path):
        """Determines the right model class for the given model file and returns an instance of that class."""
        model_path = Path(model_path)
        if model_path.suffix == '.pt':
            # Determine the specific model type (e.g., RTDETR, SAM) based on file name or content
            if 'rtdetr' in model_path.stem.lower():
                return RTDETRModel(model_path)
            elif 'sam' in model_path.stem.lower():
                return SAMModel(model_path)
            else:
                raise ValueError(f"Unsupported model type for file: {model_path}")
        else:
            raise ValueError("Unsupported file extension. Only '.pt' files are supported.")
        
```

Reply:
@glenn-jocher 
> yeah that's interesting. How would the training process work for YOLO-World? Would it be YOLO training without any modifications to Clip?

well I haven't walked through the training code, but since it's got clip involved, the dataset could include text form, which means we will get a new dataloader for yolo-world training, ends up with a new trainer for it.